### PR TITLE
Build Python 3.12 wheels

### PR DESCRIPTION
**Pull request by madig**

### CODE REVIEW

**Review by anthrotype — APPROVED** [review]

**Comment by HinTak** [comment]
I have unblocked CI. At the moment the only likely thing is timing (some stages might push over 6 hours per job). Let's see if CI catches anything else.

**Comment by HinTak** [comment]
CI failed. It seems that github's python action does not support 3.12 yet. Part of the log:

```
Build wheels on ubuntu-22.04 (aarch64) for cp312
failed 48 minutes ago in 49s
2s
12s
15s
5s
11s
Run pypa/cibuildwheel@v2.[1](https://github.com/kyamagu/skia-python/actions/runs/6395333786/job/17366861532#step:5:1)[2](https://github.com/kyamagu/skia-python/actions/runs/6395333786/job/17366861532#step:5:2).1
  with:
    package-dir: .
    output-dir: wheelhouse
  env:
    CIBW_BUILD: cp[3](https://github.com/kyamagu/skia-python/actions/runs/6395333786/job/17366861532#step:5:3)12-*
    CIBW_SKIP: *musllinux*
    CIBW_ARCHS: aarch6[4](https://github.com/kyamagu/skia-python/actions/runs/6395333786/job/17366861532#step:5:4)
    CIBW_ENVIRONMENT_MACOS: TARGET_ARCH=aarch[6](https://github.com/kyamagu/skia-python/actions/runs/6395333786/job/17366861532#step:5:6)4
    CIBW_BEFORE_ALL: bash scripts/build_Linux.sh
    CIBW_BEFORE_BUILD: pip install pybind11 numpy
    CIBW_TEST_REQUIRES: pytest pillow glfw
    CIBW_TEST_REQUIRES_MACOS: pytest pillow pyopengl
    CIBW_TEST_COMMAND: python -m pytest {project}/tests
    CIBW_TEST_COMMAND_LINUX: xvfb-run -s "-screen 0 640x480x24" python -m pytest {project}/tests
  
    CIBW_TEST_SKIP: *-macosx_arm64
Run actions/setup-python@v4
  with:
    python-version: 3.[7](https://github.com/kyamagu/skia-python/actions/runs/6395333786/job/17366861532#step:5:7) - 3.[11](https://github.com/kyamagu/skia-python/actions/runs/6395333786/job/17366861532#step:5:11)
    update-environment: false
    check-latest: false
    token: ***
    allow-prereleases: false
...
cibuildwheel: No build identifiers selected: BuildSelector(build_config='cp312-*', skip_config='*musllinux*', requires_python=None, prerelease_pythons=False)
Error: Process completed with exit code 3.
```

See the line after `actions/setup-python@v4`.

**Comment by HinTak** [comment]
Okay, python 3.12 was released today - so it is not available anywhere yer.

**Comment by anthrotype** [comment]
cibuildwheel installs its own pythons, I think you just need to make sure cibuildwheel is up to date, it definitely can build 3.12 wheels

**Comment by HinTak** [comment]
CI failed. So this cannot be merged (yet).

**Comment by HinTak** [comment]
[logs_61.zip](https://github.com/kyamagu/skia-python/files/12797262/logs_61.zip)

CI log. Please fix

**Comment by kyamagu** [comment]
Yes, this requires the up-to-date cibuildwheel
https://github.com/pypa/cibuildwheel/releases

File: .github/workflows/ci.yml
Comment (HinTak) [inline .github/workflows/ci.yml:1]: This line needs to be 2.16.2

Comment (HinTak) [reply .github/workflows/ci.yml:1]: The line which says 'pypa/cibuildwheel@v2.12.1'  around line 120.

**Review by HinTak — APPROVED** [review]
CI finished

**Pull request merged**